### PR TITLE
added the ability to use potions & revive on pokemon

### PR DESCRIPTION
**Pull request by ramarro123**
I am adding the ability to use potion & revive for pokemons.

there are also 2 utility method on item class to do some sanity check.

Note that for now, seems that protobuf have an issue
https://github.com/AeonLucid/POGOProtos/issues/72

but when it's done, we can proceed with that part.

waiting for gym :D

### CODE REVIEW

**Comment by Jari27** [comment]
Can you use proper punctuation and capitalisation for the javadoc?

**Comment by Grover-c13** [comment]
Did this pass checkstyle?

**Comment by ramarro123** [comment]
no, sorry, it doesn't, i just recheck :( let me update it

**Comment by ramarro123** [comment]
ok, checkstyle now it's correct, sorry, i am having hard time to get into that :) i'll learn..eventually...

File: src/main/java/com/pokegoapi/api/inventory/Item.java [hunk 1]
```diff
@@ -39,6 +39,11 @@ public boolean isUnseen() {
 		return proto.getUnseen();
 	}
 
+	/**
+	 * Check if the item it's a potion
```
Comment (FabianTerhorst) [inline src/main/java/com/pokegoapi/api/inventory/Item.java:43]: missing dot

File: src/main/java/com/pokegoapi/api/inventory/Item.java [hunk 2]
```diff
@@ -47,6 +52,11 @@ public boolean isPotion() {
 				;
 	}
 
+	/**
+	 * Check if the item it's a revive
```
Comment (FabianTerhorst) [inline src/main/java/com/pokegoapi/api/inventory/Item.java:56]: missing dot

File: src/main/java/com/pokegoapi/api/pokemon/Pokemon.java [hunk 3]
```diff
@@ -379,7 +378,7 @@ public double getBaseFleeRate() {
 	}
 
 	/**
-	 * check if pokemon its injured but not fainted. need potions to heal
+	 * Check if pokemon its injured but not fainted. need potions to heal
```
Comment (FabianTerhorst) [inline src/main/java/com/pokegoapi/api/pokemon/Pokemon.java:398]: dot

File: src/main/java/com/pokegoapi/api/pokemon/Pokemon.java [hunk 4]
```diff
@@ -397,11 +396,13 @@ public boolean isFainted() {
 	}
 
 	/**
-	 * Heal a pokemon, using various fallback for potion
+	 * Heal a pokemon, using various fallbacks for potions
```
Comment (FabianTerhorst) [inline src/main/java/com/pokegoapi/api/pokemon/Pokemon.java:416]: dot

File: src/main/java/com/pokegoapi/api/pokemon/Pokemon.java [hunk 5]
```diff
@@ -450,11 +454,13 @@ public boolean isFainted() {
 	}
 
 	/**
-	 * Heal a pokemon, using various fallback for potion
+	 * Revive a pokemon, using various fallbacks for revive items
```
Comment (FabianTerhorst) [inline src/main/java/com/pokegoapi/api/pokemon/Pokemon.java:477]: dot

**Comment by FabianTerhorst** [comment]
The method documentation title has to end with a dot.

**Comment by FabianTerhorst** [comment]
Im not sure why checkstyle doesn´t find everything.

**Comment by ramarro123** [comment]
checkstyle was modified
https://github.com/Grover-c13/PokeGOAPI-Java/pull/226

and merged...

Remove the rule that force to write a dot at the end of the firstline of javadoc.

so, i guess that this constrain was removed & accepted

**Comment by FabianTerhorst** [comment]
The removed items in the xml aren´t for this. Just add the dot like it is on all other methods.

**Comment by vmarchaud** [comment]
The dot isnt necessary @FabianTerhorst now
Thanks for your contribution @ramarro123 it will be merged soon.

**Comment by ramarro123** [comment]
@vmarchaud  can you please review the last modify that i push  my repo?

i am not sure on the approach, i basically copy the item paradigm.

pokemon stamina get returned from the heal stuff, so i want to update it back without having to refresh playerInfo.

On item, it's the same for count.

i copy the variable, use the annotation and on my side work, but i prefer a second opinion on that.

tnx

**Comment by Grover-c13** [comment]
any chance you could resolve current conflicts and ill merge :P?

**Comment by ramarro123** [comment]
let me know if everything it's fine

**Comment by Grover-c13** [comment]
In future could you do your commits in english too please :P

**Pull request merged**